### PR TITLE
74965 - Interpret uniform messages on 400

### PR DESCRIPTION
**Pull request by kfaln**
Parse input and business validation error messages.

### CODE REVIEW

File: src/modules/Preservation/http/PreservationApiClient.ts [hunk 1]
```diff
@@ -428,6 +428,22 @@ function getPreservationApiError(error: AxiosError): PreservationApiError {
     if (error.response.status == 404) {
         return new PreservationApiError(error.response.data, error.response);
     }
+    if (error.response.status == 400) {
+        try {
+            // input and business validation errors
+            let validationErrorMessage = error.response.data.title;
```
Comment (Cnordbo) [inline src/modules/Preservation/http/PreservationApiClient.ts:434]: No sure if this gives us any relevant context ("validation error occured" or " Bussiness validation errors: X " )

File: src/modules/Preservation/http/PreservationApiClient.ts [hunk 2]
```diff
@@ -428,6 +428,22 @@ function getPreservationApiError(error: AxiosError): PreservationApiError {
...
+            const validationErrors = error.response.data.errors;
+
+            for (const validatedField in validationErrors) {
+                const fieldErrors = validationErrors[validatedField].join(' | ');
+                validationErrorMessage += ` ${validatedField}: ${fieldErrors} `;
```
Comment (Cnordbo) [inline src/modules/Preservation/http/PreservationApiClient.ts:439]: the field doesnt necessarily give the user any context, currently we should probably rely on descriptive errormessage as fieldnames are internal to backend.

File: src/modules/Preservation/http/PreservationApiClient.ts (hunk 1 above)
Comment (kfaln) [reply src/modules/Preservation/http/PreservationApiClient.ts:434]: I guess the back-end should provide more context-specific messages, if the need arises. Front-end will display whatever it comes up with.

File: src/modules/Preservation/http/PreservationApiClient.ts (hunk 2 above)
Comment (kfaln) [reply src/modules/Preservation/http/PreservationApiClient.ts:439]: Perhaps, but I think you will get a pretty close match on the front VS back field name most of the times. Here is an example:

Remark: The length of 'Remark' must be 255 characters or fewer. You entered 339 characters.

Comment (kfaln) [reply src/modules/Preservation/http/PreservationApiClient.ts:439]: The field is also a part of the error message, so again it's up to back-end to supply niceness to the user.

Comment (Cnordbo) [reply src/modules/Preservation/http/PreservationApiClient.ts:439]: The field in the error message is handled by the implementor as a string error message. 
The field ( as a key ) is just the property of an object. 

ex. "StepId: Step does not exist" 
ex. "RequirementDescription: Description can not be more then 255 characters"

Comment (kfaln) [reply src/modules/Preservation/http/PreservationApiClient.ts]: Removed the internal field name from error message.

**Pull request merged**